src/grpc/projects.rs

```diff
@@ -105,328 +109,188 @@ impl ProjectService for ProjectServiceImpl {
-        // })?;
-
-        // let inner_request = request.into_inner();
-        // let object_id = tonic_invalid!(
-        //     DieselUlid::from_str(&inner_request.project_id),
-        //     "ULID conversion error"
-        // );
-        // let ctx = Context::ResourceContext(ResourceContext::Project(Some(ApeResourcePermission {
-        //     id: object_id,
-        //     level: PolicyLevels::WRITE, // append?
-        //     allow_sa: false,
-        // })));
-
-        // match &self.authorizer.check_permissions(&token, ctx) {
-        //     Ok(b) => {
-        //         if *b {
-        //             // ToDo!
-        //             // PLACEHOLDER!
-        //             DieselUlid::generate()
-        //         } else {
-        //             return Err(tonic::Status::permission_denied("Not allowed."));
-        //         }
-        //     }
-        //     Err(e) => {
-        //         log::debug!("{}", e);
-        //         return Err(tonic::Status::permission_denied("Not allowed."));
-        //     }
-        // };
-        // let client = self.database.get_client().await.map_err(|e| {
-        //     log::error!("{}", e);
-        //     tonic::Status::unavailable("Database not avaliable.")
-        // })?;
-        // Object::update_name(object_id, inner_request.name, &client)
-        //     .await
-        //     .map_err(|e| {
-        //         log::error!("{}", e);
-        //         tonic::Status::aborted("Database update failed.")
-        //     })?;
-        // let object = Object::get_object_with_relations(&object_id, &client)
-        //     .await
-        //     .map_err(|e| {
-        //         log::error!("{}", e);
-        //         tonic::Status::aborted("Database update failed.")
-        //     })?;
-        // let project = Some(object.try_into().map_err(|e| {
-        //     log::error!("{}", e);
-        //     tonic::Status::aborted("Database request failed.")
-        // })?);
-        // Ok(tonic::Response::new(UpdateProjectNameResponse { project }))
+        log_received!(&request);
+
+        let token = tonic_auth!(
+            get_token_from_md(request.metadata()),
+            "Token authentication error."
+        );
+
+        let request = NameUpdate::Project(request.into_inner());
+        let collection_id = tonic_invalid!(request.get_id(), "Invalid project id");
+        let ctx = Context::res_ctx(collection_id, DbPermissionLevel::WRITE, true);
+
+        tonic_auth!(
+            self.authorizer.check_permissions(&token, vec![ctx]).await,
+            "Unauthorized"
+        );
+
+        let project = tonic_internal!(
+            self.database_handler.update_name(request).await,
+            "Internal database error."
+        );
+        self.cache
+            .update_object(&project.object.id, project.clone());
+        let project: generic_resource::Resource =
+            tonic_internal!(project.try_into(), "Collection conversion error");
```

Comment:
Copy & paste only partially adapted.

```suggestion
        let project_id = tonic_invalid!(request.get_id(), "Invalid project id");
        let ctx = Context::res_ctx(project_id, DbPermissionLevel::WRITE, true);

        tonic_auth!(
            self.authorizer.check_permissions(&token, vec![ctx]).await,
            "Unauthorized"
        );

        let project = tonic_internal!(
            self.database_handler.update_name(request).await,
            "Internal database error."
        );
        self.cache
            .update_object(&project.object.id, project.clone());
        let project: generic_resource::Resource =
            tonic_internal!(project.try_into(), "Project conversion error");
```